backup-stores/azure/src/main/java/io/camunda/zeebe/backup/azure/ManifestManager.java

```diff
@@ -161,6 +161,20 @@ void markAsFailed(final BackupIdentifier manifestId, final String failureReason)
     }
   }
 
+  public void deleteManifest(final BackupIdentifier id) {
+    final BlobClient blobClient = blobContainerClient.getBlobClient(manifestIdPath(id));
+    final Manifest manifest = getManifest(id);
+    if (manifest == null) {
+      return;
+    } else if (manifest.statusCode() == StatusCode.IN_PROGRESS) {
+      throw new UnexpectedManifestState(
+          "Cannot delete Backup %s while saving is in progress.".formatted(id.toString()));
+    }
+    if (blobClient.exists()) {
+      blobClient.delete();
+    }
```

Comment:
:x: You already checked that it exists by calling `getManifest` and checking for null. No need to make another request to check for existance, just delete it here.